src/bluesky/utils/__init__.py

```diff
@@ -1914,3 +1915,49 @@ async def maybe_await(ret: SyncOrAsync[T]) -> T:
         # Mypy does not understand how to narrow type to non-awaitable in this
         # instance, see https://github.com/python/mypy/issues/15520
         return ret  # type: ignore
+
+
+class Plan:
+    def __init__(self, f, *args, **kwargs) -> None:
+        self._iter = f(*args, **kwargs)
+        self._stack = traceback.format_stack()
+        self._stack = self._stack[:-2]
+        self._stack += [
+            f"RuntimeWarning: plan `{f.__name__}` was never iterated" ", did you mean to use `yield from`?"
+        ]
+
+    def __iter__(self):
+        self._stack = None
+        return (yield from self._iter)
+
+    def __next__(self):
```

Comment:
Why do we need `__next__`?  Do we also need to forward `.send` and `.throw`?

Reply:
The test_repeater calls next() on a plan: https://github.com/bluesky/bluesky/blob/add624e1b7f03c9b4b981e78dfc2333ed01e1f0c/src/bluesky/tests/test_new_examples.py#L641

`.send` and `.throw` would get forwarded here:
https://github.com/bluesky/bluesky/blob/3173c97e8017892a9f0a1f5c57a0661ed92a4b7b/src/bluesky/utils/__init__.py#L1941-L1942
Which would behave transparently as if it were a real plan, unless you want me to explicitly forward them

Reply:
I think this is conflating an Iterator and an Iterable. A plan should not implement `__next__`.

```py
In [1]: def g():
   ...:     yield 1
   ...:     yield 2
   ...: 

In [2]: next(g)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[2], line 1
----> 1 next(g)

TypeError: 'function' object is not an iterator

In [3]: gi = g()

In [4]: next(gi)
Out[4]: 1
```

Reply:
^ That comment might be overconfident. I should take more time to dig into this if need be, but something smells suspicious about implementing both `__iter__` and `__next__` on this object.

Reply:
Ah, I somehow missed `__getattr__` being added, it is probably better if we explicitly forward the ones we need.

Reply:
Explicit forwarding of `send()` and `throw()` added in https://github.com/bluesky/bluesky/pull/1709/commits/1b981b7e34958cb3285fb709b14c8b792d1cc5ea

Reply:
I now understand what I was confused about.